src/com/oltpbenchmark/api/Worker.java

```diff
@@ -321,6 +335,8 @@ public final void run() {
                         latencies.addLatency(type.getId(), start, end, this.id
                                 , phase.id);
                         intervalRequests.incrementAndGet();
+                        long latent = ((end - start + 500) / 1000);
```

Comment:
this is a truncation error, since you don't need to divide until you are producing the averages you should postpone scaling the time factors until the last possible moment. Otherwise you will accumulate small errors and the longer the test runs, the more inaccurate your averages will be.